### ISSUE TITLE
[filterscripts] SimulationFiltering.py may fail due to missing key (Trac #2029)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2021]
Hi,

while running my simulation, I have seen this error:

NOTICE (HeseFollowup): injecting followup message into frame for run 1, event 639 (hesefollowup.py:129 in generate_small_HESE_followup_message)
ERROR (PythonFunction): Error running python function as module: (PythonFunction.cxx:173 in virtual void PythonFunction::Process())
ERROR (I3Module): HESEFollowup_send_followup: Exception thrown (I3Module.cxx:116 in void I3Module::Do(void (I3Module::*)()))
Traceback (most recent call last):
  File "/scratch/condor/dir_597/condor_exec.exe", line 418, in <module>
    main(opts)
  File "/scratch/condor/dir_597/condor_exec.exe", line 379, in main
    tray.Execute()
  File "/data/user/flauber/software/icerec_V05-00-05/build/lib/I3Tray.py", line 234, in Execute
    super(I3Tray, self).Execute()
  File "/data/user/flauber/software/icerec_V05-00-05/build/lib/icecube/filterscripts/hesefollowup.py", line 178, in generate_small_HESE_followup_message
    'mpe_muex'      : round(frame["OnlineL2_BestFit_MuEx"].energy, 5),
KeyError: 'OnlineL2_BestFit_MuEx'
Removing cpandel parameterization ... done
WARN (I3BasicSeedService): (OnlineL2_multitseedprep) In 255 SetEvent() calls, 1 first guess track(s) were found but had a fit status that was not OK! (I3BasicSeedService.c$

It only happened for one job out of about 100, still I think it should not happen (and as the SimlationFiltering.py is self contained I hope its not an error due to my doing)

Migrated from https://code.icecube.wisc.edu/ticket/2029
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:14:38", 
    "description": "Hi,\n\nwhile running my simulation, I have seen this error:\n\nNOTICE (HeseFollowup): injecting followup message into frame for run 1, event 639 (hesefollowup.py:129 in generate_small_HESE_followup_message)\nERROR (PythonFunction): Error running python function as module: (PythonFunction.cxx:173 in virtual void PythonFunction::Process())\nERROR (I3Module): HESEFollowup_send_followup: Exception thrown (I3Module.cxx:116 in void I3Module::Do(void (I3Module::*)()))\nTraceback (most recent call last):\n  File \"/scratch/condor/dir_597/condor_exec.exe\", line 418, in <module>\n    main(opts)\n  File \"/scratch/condor/dir_597/condor_exec.exe\", line 379, in main\n    tray.Execute()\n  File \"/data/user/flauber/software/icerec_V05-00-05/build/lib/I3Tray.py\", line 234, in Execute\n    super(I3Tray, self).Execute()\n  File \"/data/user/flauber/software/icerec_V05-00-05/build/lib/icecube/filterscripts/hesefollowup.py\", line 178, in generate_small_HESE_followup_message\n    'mpe_muex'      : round(frame[\"OnlineL2_BestFit_MuEx\"].energy, 5),\nKeyError: 'OnlineL2_BestFit_MuEx'\nRemoving cpandel parameterization ... done\nWARN (I3BasicSeedService): (OnlineL2_multitseedprep) In 255 SetEvent() calls, 1 first guess track(s) were found but had a fit status that was not OK! (I3BasicSeedService.c$\n\nIt only happened for one job out of about 100, still I think it should not happen (and as the SimlationFiltering.py is self contained I hope its not an error due to my doing)\n\n\n", 
    "reporter": "flauber", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067278746682", 
    "component": "combo reconstruction", 
    "summary": "[filterscripts] SimulationFiltering.py may fail due to missing key", 
    "priority": "minor", 
    "keywords": "filterscripts", 
    "time": "2017-05-22T10:06:09", 
    "milestone": "", 
    "owner": "claudio.kopper", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170522T16:28:19: 
* david.schultz changed summary from "SimulationFiltering.py may fail due to missing key" to "[filterscripts] SimulationFiltering.py may fail due to missing key"
* david.schultz commented:

> This seems to be an error in the hesefollowup, not checking if the key is valid first.  I note that the GFU and OFU filters both check if the key is valid.
> 
> There are a few cases where the hese followup can be run while MuEx is not run.  It's not likely, but simulation is good for finding corner cases.


* david.schultz changed owner from "" to "claudio.kopper"
* david.schultz changed component from "simulation" to "icerec"
* david.schultz changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20170818T16:02:15: david.schultz changed milestone from "" to "September 2017"

**Comment by icecube-trac** [comment]
Trac update at 20171012T19:04:49: 
* david.schultz changed status from "assigned" to "closed"
* david.schultz changed resolution from "" to "fixed"
* david.schultz commented:

> In changeset 158641:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="158641"
> make mpe_muex optional, since it might not always be generated. fixes #2021
> ```

**Comment by icecube-trac** [comment]
Trac update at 20171012T20:18:36: thomas.kintscher commented:

> In changeset 158645:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="158645"
> Fixes #2021, without changing the contents of the short followup message.
> I could only see MuEx missing in the frame when
> not even a PoleMuonLlhFit (until 2016) or PoleMuonLinefit (2017 filtering)
> has converged.
> In that case the other recos, such as SplineMPE, shouldn't be trusted either.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:14:38: 
* olivas commented:

> Milestone September 2017 deleted


* olivas changed milestone from "September 2017" to ""
